src/main/java/uk/gov/homeoffice/digital/sas/timecard/producers/KafkaProducerTimeEntry.java

```diff
@@ -0,0 +1,53 @@
+package uk.gov.homeoffice.digital.sas.timecard.producers;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.kafka.core.KafkaTemplate;
+import org.springframework.kafka.support.SendResult;
+import org.springframework.stereotype.Component;
+import org.springframework.util.concurrent.ListenableFuture;
+import org.springframework.util.concurrent.ListenableFutureCallback;
+import uk.gov.homeoffice.digital.sas.timecard.enums.KafkaAction;
+import uk.gov.homeoffice.digital.sas.timecard.model.KafkaEventMessage;
+import uk.gov.homeoffice.digital.sas.timecard.model.TimeEntry;
+
+@Component
+@Slf4j
+public class KafkaProducerTimeEntry {
+
+  private final KafkaTemplate<String, KafkaEventMessage> kafkaTimeEntryTemplate;
+
+  public KafkaProducerTimeEntry(KafkaTemplate<String, KafkaEventMessage> kafkaTimeEntryTemplate) {
+    this.kafkaTimeEntryTemplate = kafkaTimeEntryTemplate;
+  }
+
+  public void sendMessage(TimeEntry timeEntry, KafkaAction action) {
+    KafkaEventMessage kafkaEventMessage = new KafkaEventMessage(timeEntry, action);
+    ListenableFuture<SendResult<String, KafkaEventMessage>> future = kafkaTimeEntryTemplate.send(
+        "callisto-timecard",
+        timeEntry.getOwnerId().toString(),
```

Comment:
I spoke to the architects about this and we agreed that although the owner ID may not be unique for different tenants, it is unique enough because it guarantees that the same users will end up in the same partitions